scripts/logbook.py

```diff
@@ -198,6 +124,19 @@ def write_record(content,dest_file):
     body = {'valueInputOption' : "USER_ENTERED", 'data' : data}
     result = service.spreadsheets().values().batchUpdate(spreadsheetId=spreadsheetId, body=body).execute()
 
+# Fetch UDF details
+def get_details(record_instrument,pro):
+    udf_detail=[]
+    for item in record_instrument.get("details", []):
+        if item == "Processname":
+            udf_detail.append(pro.type.name)
+        else:
+            try:
+                udf_detail.append(item+":"+pro.udf[item])
```

Comment:
We could make use of the `udf.get` method here and avoid using `try` 😉 . In that way we could also combine it as  `if.. elif..` block. I mean like

```
def get_details(record_instrument,pro):
    udf_detail = []
    for item in record_instrument.get("details", []):
        if item == "Processname":
            udf_detail.append(pro.type.name)
        elif pro.udf.get(item):
            udf_detail.append(item+":"+pro.udf[item])
    return ",".join(udf_detail) if udf_detail else "-"
```